File: demos/client/src/components/Airleg.js
import React from 'react';
import {Icon} from 'semantic-ui-react';
import {diffFormatted, formatTime} from '../util/dates';

const Airleg = ({departureDate, arrivalDate, departureAirport, arrivalAirport, segments}) => {
    let departureTime = formatTime(departureDate);
    let arrivalTime = formatTime(arrivalDate);
    let totalTime = diffFormatted(departureDate, arrivalDate);
    return(
        <p>
            {departureTime} &emsp; <strong>{departureAirport.iata}</strong> 
            &emsp;<Icon name="long arrow right"/>&emsp; 
            {arrivalTime} &emsp; <strong>{arrivalAirport.iata}</strong> &emsp;
            <span className='gray'>{totalTime}</span>&emsp;
            <span className='gray'>{segments.length-1} stops</span>&emsp;
            <span className='gray'>({getStopsIata(segments)})</span>
        </p>
    );
}

function getStopsIata(segments){
    var iatas = "";
    segments.forEach((x)=>iatas += "-"+x.arrivalAirport.iata)
    return iatas;
}

export default Airleg;

File: farandula-ruby/lib/farandula/air_leg.rb
module Farandula
  class AirLeg
      attr_accessor :id
      attr_accessor :departure_airport_code
      attr_accessor :departure_date
      attr_accessor :arrival_airport_code
      attr_accessor :arrival_date
      attr_accessor :segments
      attr_accessor :price

      def initalize(
        id                      = nil,
        departure_airport_code  = nil,
        departure_date          = nil,
        arrival_airport_code    = nil,
        arrival_date            = nil,
        segments                = [],
        price                   = nil
      )
        
        @id                     = id
        @departure_airport_code = departure_airport_code
        @departure_date         = departure_date
        @arrival_airport_code   = arrival_airport_code
        @arrival_date           = arrival_date
        @segments               = segments
        @price                  = price

      end
  end
  
end